# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/base/AppShutdown.cpp
# Commit: 6450088b6b73
# Full Hash: 6450088b6b73ffc17c79cd7097b1bfe30d00e207
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-09-14 21:29:41
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1656261 - Back out all recent StartupCache work r=RyanVM
#   
#   This backs out all work from bug 1627075 as well as all of its
#   descendents. There were a few conflicts when backing this out but
#   overall it was pretty clean, so I would say it's a fairly mild
# ==============================================================================

diff -r b0fabcd60430 -r 6450088b6b73 xpcom/base/AppShutdown.cpp
--- a/xpcom/base/AppShutdown.cpp	Mon Sep 14 17:32:40 2020 +0000
+++ b/xpcom/base/AppShutdown.cpp	Mon Sep 14 17:00:53 2020 +0000
@@ -142,7 +142,7 @@
 
   // Very early shutdowns can happen before the startup cache is even
   // initialized; don't bother initializing it during shutdown.
-  if (auto* cache = scache::StartupCache::GetSingleton()) {
+  if (auto* cache = scache::StartupCache::GetSingletonNoInit()) {
     cache->MaybeInitShutdownWrite();
   }
 }
@@ -152,7 +152,7 @@
   // the late write checking code. Anything that writes to disk and which
   // we don't want to skip should be listed out explicitly in this section.
   if (aPhase == sFastShutdownPhase || aPhase == sLateWriteChecksPhase) {
-    if (auto* cache = scache::StartupCache::GetSingleton()) {
+    if (auto* cache = scache::StartupCache::GetSingletonNoInit()) {
       cache->EnsureShutdownWriteComplete();
     }
 